Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1101,6 +1101,22 @@ public extension Purchases {
 
 #endif
 
+    @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+    func handleObserverModeTransaction(
+        purchaseResult: StoreKit.Product.PurchaseResult
+    ) async throws -> StoreTransaction? {
+        guard !self.systemInfo.observerMode else {
+            throw NewErrorUtils.observerModeNotEnabledError()
```

Comment:
Throw early, throw often. 😄

Reply:
I would prevent creating yet another error for this. That's yet another error that users need to handle not just on this API, but in **all** of our APIs, and maybe localize to different languages, etc. 
I would instead throw `unsupportedError` or `configurationError`.